src/com/owncloud/android/services/observer/InstantUploadsObserverBasedOnINotify.java

```diff
@@ -0,0 +1,194 @@
+/**
+ *   ownCloud Android client application
+ *
+ *   @author David A. Velasco
+ *   Copyright (C) 2017 ownCloud GmbH.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License version 2,
+ *   as published by the Free Software Foundation.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.owncloud.android.services.observer;
+
+import android.content.Context;
+import android.os.FileObserver;
+import android.os.Handler;
+import android.os.Looper;
+
+import com.owncloud.android.db.PreferenceManager.InstantUploadsConfiguration;
+import com.owncloud.android.lib.common.utils.Log_OC;
+
+import java.util.HashMap;
+
+/**
+ * Observer watching a folder to request the upload of new pictures or videos inside it.
+ *
+ * Implementation based in {@link FileObserver}, class in the Android framework acting as a hook
+ * with iNotify ({@see https://en.wikipedia.org/wiki/Inotify}).
+ */
+public class InstantUploadsObserverBasedOnINotify extends FileObserver implements InstantUploadsObserver {
+
+    private static final String TAG = InstantUploadsObserverBasedOnINotify.class.getSimpleName();
+
+    private static final int CREATE_MASK = (
+            FileObserver.CREATE | FileObserver.MODIFY | FileObserver.CLOSE_WRITE |
+            FileObserver.MOVED_TO
+    );
+
+    private static final int ALL_EVENTS_EVEN_THOSE_NOT_DOCUMENTED = 0x7fffffff;   // NEVER use 0xffffffff
```

Comment:
Why not use 0xffffffff? If you suggest not to use it, a comment explaining the reason could be interesting.